src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -785,6 +795,51 @@ public static unsafe SocketError SendAsync(SafeCloseSocket handle, IList<ArraySe
             return errorCode;
         }
 
+        // This assumes preBuffer/postBuffer are pinned already 
+
+        private static unsafe SocketError TransmitFileHelper(
+            SafeHandle socket, 
+            SafeHandle fileHandle,
+            SafeHandle overlapped,
+            byte[] preBuffer,
+            byte[] postBuffer,
+            TransmitFileOptions flags)
+        {
+            bool needTransmitFileBuffers = false;
+            Interop.Mswsock.TransmitFileBuffers transmitFileBuffers;
```

Comment:
I'm not sure if this is guaranteed to be nulled out.  If it's not, and only one of preBuffer or postBuffer is provided, you could end up passing garbage for Head/HeadLength or Tail/TailLength.  I'd suggest changing this to:
```C#
var transmitFileBuffers = default(Interop.Mswsock.TransmitFileBuffers);
```